src/neural/cuda/network_cudnn.cc

```diff
@@ -1078,6 +1078,17 @@ std::unique_ptr<Network> MakeCudnnNetwork(const std::optional<WeightsFile>& w,
           pblczero::NetworkFormat::NETWORK_CLASSICAL_WITH_HEADFORMAT &&
       weights.format().network_format().network() !=
           pblczero::NetworkFormat::NETWORK_SE_WITH_HEADFORMAT) {
+#ifdef PLAIN_CUDA
+    // Check if we must switch to cuda.
+    if (weights.format().network_format().network() ==
+        pblczero::NetworkFormat::NETWORK_ATTENTIONBODY_WITH_HEADFORMAT) {
+      CERR << "Nework format not supported directly, switching backend.";
+      return NetworkFactory::Get()->Create(
+          "cuda" +
+              std::string(std::is_same<half, DataType>::value ? "-fp16" : ""),
+          weights, options);
+    }
+#endif
```

Comment:
Maybe add #else with the exception message that adds more context? ("Please download/use CUDA version/backend instead")

Reply:
I decided to simplify things, now the cuda backend will always be available when cudnn is built.

Reply:
Is it guaranteed that when libraries needed for the cudnn backend are present on the system, libraries for cuda are also there? (I guess so, likely cuda is subset of cudnn, just confirming)

If not, it makes sense to allow building just cudnn backend without cuda.

Reply:
Yes, both backends use the same cuda libraries (cudart and cublas).
There is one thing I forgot to mention, if the cudnn lib is not found you will still get the cuda backend even if `plain_cuda=false`. I did it like that to simplify the logic in `meson.build` but I'm having second thoughts.